src/intersections/lines.jl

```diff
@@ -9,20 +9,24 @@ The intersection type can be one of three types:
 2. overlap at more than one point
 3. do not overlap nor intersect
 =#
-function intersection(f, l1::Line{3,T}, l2::Line{3,T}) where {T}
+function intersection(f::Function, l1::Line{3,T}, l2::Line{3,T}) where {T}
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
 
-  if measure(Tetrahedron(a, b, c, d)) > 0
+  if measure(Tetrahedron(a, b, c, d)) > 0 # not in same plane
     return @IT NoIntersection nothing f
-  elseif isapprox(abs((b - a) × (c - d)), zero(T), atol=atol(T)^2)
-    return @IT OverlappingLines l1 f
+  elseif isapprox(norm((b - a) × (c - d)), zero(T), atol=atol(T)^2)
```

Comment:
in 3D the cross product returns a vector so `abs` yielded an error here

Reply:
Interesting. Our tests are not covering this case then? We should add tests to make sure that this branch is covered.

Reply:
add the following test case for 3D line intersections: `intersect(Line(1., 0, 0), Line(0., 1, 0))`

Reply:
You need to add the tests in the PR so that GitHub will run them and check correctness. We cannot merge this without tests included. Can you please add the tests to the PR?

Reply:
ok, I will do, so I just add test cases in the `/test/intersections.jl` file?
sorry for the inconvenience - first PR

Reply:
No worries, the first PR is always a learning curve :) Yes, you can add tests to the corresponding file in the test folder, in this case intersections.jl. There is a specific section for line-line intersection I think. If not, feel free to add one.

Reply:
Cool :)
I added 4 tests for 3d lines at the correct place in the test/intersections.jl